### PR TITLE
Amcl refactorization

### PR DESCRIPTION
**Pull request by mhpanah**
In this PR strutils is factored out and placed into nav2_util to address issue #61.

Also, AMCLLaser, AMCLOdom, and AMCLSensor classes have been renamed to Laser, Odom, and Sensor respectively.

### CODE REVIEW

File: src/libs/nav2_util/include/nav2_util/sensors/laser.h [hunk 1]
```diff
@@ -29,7 +29,7 @@
 #ifndef AMCL_LASER_H
```
Comment (crdelsey) [inline src/libs/nav2_util/include/nav2_util/sensors/laser.h]: Have you run the linters on this change? I'm thinking AMCL_LASER_H will fail cpplint.

File: src/libs/nav2_util/include/nav2_util/strutils.hpp [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+#include <string>
+
+class strutils
+{
+public:
+  static std::string stripLeadingSlash(const std::string & in);
+};
+
+std::string strutils::stripLeadingSlash(const std::string & in)
```
Comment (mjeronimo) [inline src/libs/nav2_util/include/nav2_util/strutils.hpp]: You could have just defined the function in the class declaration above.

File: src/libs/nav2_util/include/nav2_util/sensors/laser.h (hunk 1 above)
Comment (mhpanah) [reply src/libs/nav2_util/include/nav2_util/sensors/laser.h]: There will be another PR to address all the linter issues.

**Comment by SteveMacenski** [comment]
Why was AMCL split up into multiple packages and now renamed without the AMCL tag? It's no longer clear in my opinion to a new 3rd party what the odom, laser, or sensor classes mean anymore without the AMCL tag or in the AMCL package itself.

**Comment by mhpanah** [comment]
@SteveMacenski The main reason is code reusability.  Our long term plan is to be able to support different localization methods.  Most likely other localization algorithms will use similar sensors. So it would be beneficial to have generic sensor, odom, and laser classes and also generic kf, ukf, pf, etc. filtering.  As of now these classes are just renamed and placed in the util directory and to make them useful we need to change them substantially.

**Pull request merged**